backend/tracim_backend/applications/upload_permissions/email_manager.py

```diff
@@ -0,0 +1,242 @@
+        context = {
+            "receiver": receiver,
+            "workspace": workspace,
+            "uploader_username": uploader_username,
+            "uploaded_contents": uploaded_contents,
+        }
+        body_html = self._render_template(
+            mako_template_filepath=html_template_file_path, context=context, translator=translator
+        )
+
+        part2 = MIMEText(body_html, "html", "utf-8")
+        # Attach parts into message container.
+        # According to RFC 2046, the last part of a multipart message,
+        # in this case the HTML message, is best and preferred.
+        message.attach(part2)
+        return message
+
+    def notify_upload_permission(
+        self,
+        emitter: User,
+        workspace: Workspace,
+        upload_permission_receivers: typing.List[UploadPermissionInContext],
+        upload_permission_password: str,
+    ) -> None:
+        """
+        Send mails to notify users for sharing content
+        :param emitter: User emitter of the sharing
+        :param workspace: workspace where receivers can now upload file
+        :param upload_permission_receivers: list of upload_permission
+        :param upload_permission_password: cleartext password of the sharing
+        """
+
+        email_sender = EmailSender(
+            self.config, self._smtp_config, self.config.EMAIL__NOTIFICATION__ACTIVATED
+        )
+        upload_permission_password_enabled = False
+        if upload_permission_password:
+            upload_permission_password_enabled = True
+        translator = Translator(self.config, default_lang=emitter.lang)
+        message = self._notify_emitter(
+            emitter=emitter,
+            workspace=workspace,
+            upload_permission_receivers=upload_permission_receivers,
+            upload_permission_password=upload_permission_password,
+            translator=translator,
+        )
+        send_email_through(
+            config=self.config, sendmail_callable=email_sender.send_mail, message=message
+        )
+        emails_receivers_list = [
+            upload_permission.email for upload_permission in upload_permission_receivers
+        ]
+        logger.info(
+            self,
+            'Generating upload permission mail from user "{}" to "{}"'.format(
+                emitter.user_id, "".join(emails_receivers_list)
+            ),
+        )
+        for upload_permission in upload_permission_receivers:
+            message = self._notify_receiver(
+                emitter=emitter,
+                workspace=workspace,
+                upload_permission=upload_permission,
+                upload_permission_password_enabled=upload_permission_password_enabled,
+                translator=translator,
+            )
+            send_email_through(
+                config=self.config, sendmail_callable=email_sender.send_mail, message=message
+            )
+
+    def _notify_emitter(
```

Comment:
same remark here

Reply:
issue about this there #2206
Does concern also share app, so not fixed here.